### PR TITLE
Make the best effort to clean up stale NVMe and device mapper devices

### PR DESCRIPTION
**Pull request by derekbit**
No need to error out.

#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#7608

#### What this PR does / why we need it:

Just make the best effort to clean up stale NVMe and device mapper devices.

### CODE REVIEW

File: app/cmd/start.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func cleanupStaledNvmeAndDmDevices() error {
 			logrus.Infof("Cleaning up NVMe subsystem %v: NQN %v", sys.Name, sys.NQN)
 
 			if err := helpernvme.DisconnectTarget(sys.NQN, executor); err != nil {
-				return errors.Wrapf(err, "failed to disconnect NVMe subsystem %v: NQN %v", sys.Name, sys.NQN)
+				logrus.WithError(err).Warnf("Failed to disconnect NVMe subsystem %v: NQN %v", sys.Name, sys.NQN)
```
Comment (shuo-wu) [inline app/cmd/start.go:146]: NIT: "...will continue the cleanup"

File: app/cmd/start.go [hunk 2]
```diff
@@ -152,7 +152,7 @@ func cleanupStaledNvmeAndDmDevices() error {
 			}
 			logrus.Infof("Removing dm device %v", dmDeviceName)
 			if err := helperutil.DmsetupRemove(dmDeviceName, true, true, executor); err != nil {
-				return errors.Wrapf(err, "failed to remove dm device %v", dmDeviceName)
+				logrus.WithError(err).Warnf("Failed to remove dm device %v", dmDeviceName)
```
Comment (shuo-wu) [inline app/cmd/start.go:155]: NIT: "...will continue the cleanup"

**Review by shuo-wu — COMMENTED** [review]
Should we clean up the dm device first?

**Comment by derekbit** [comment]
> Should we clean up the dm device first?

Yes. Update the order, but it is not harmful in this case. Thank you.

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**